lib/plugins/aws/package/compile/events/apiGateway/index.js

```diff
@@ -20,15 +20,187 @@ const getMethodAuthorization = require('./lib/method/authorization');
 const getMethodIntegration = require('./lib/method/integration');
 const getMethodResponses = require('./lib/method/responses');
 
+function caseInsensitive(str) {
+  return { type: 'string', regexp: new RegExp(`^${str}$`, 'i').toString() };
+}
+
+const contentHandlingSchema = { enum: ['CONVERT_TO_BINARY', 'CONVERT_TO_TEXT'] };
+
+const allowedMethods = ['GET', 'POST', 'PUT', 'PATCH', 'OPTIONS', 'HEAD', 'DELETE', 'ANY'];
+const methodPattern = new RegExp(`^(?:\\*|${allowedMethods.join('|')})$`, 'i');
+const methodPathPattern = new RegExp(`^(?:\\*|(${allowedMethods.join('|')}) (\\/\\S*))$`, 'i');
+
+const requestParametersSchema = {
+  type: 'object',
+  additionalProperties: {
+    oneOf: [
+      { type: 'boolean' },
+      {
+        type: 'object',
+        properties: {
+          required: { type: 'boolean' },
+          mappedValue: { type: 'string' },
+        },
+        additionalProperties: false,
+      },
+    ],
+  },
+};
+
+const authorizerSchema = {
+  oneOf: [
+    { type: 'string' },
+    caseInsensitive('aws_iam'),
+    { $ref: '#/definitions/awsArnString' },
+    {
+      type: 'object',
+      properties: {
+        arn: { $ref: '#/definitions/awsArn' },
+        authorizerId: { $ref: '#/definitions/awsCfInstruction' },
+        claims: { type: 'array', items: { type: 'string' } },
+        identitySource: { type: 'string' },
+        identityValidationExpression: { type: 'string' },
+        managedExternally: { type: 'boolean' },
+        name: { type: 'string' },
+        resultTtlInSeconds: { type: 'integer', minimum: 0, maximum: 3600 },
+        scopes: { type: 'array', items: { type: 'string' } },
+        type: {
+          oneOf: ['token', 'cognito_user_pools', 'request', 'aws_iam'].map(caseInsensitive),
+        },
+      },
+      required: [],
+      additionalProperties: false,
+    },
+  ],
+};
+
+const corsSchema = {
+  oneOf: [
+    { type: 'boolean' },
+    {
+      type: 'object',
+      properties: {
+        allowCredentials: { type: 'boolean' },
+        cacheControl: { type: 'string' },
+        headers: { type: 'array', items: { type: 'string' } },
+        maxAge: { type: 'integer', minimum: 1 },
+        methods: { type: 'array', items: { enum: allowedMethods } },
+      },
+      oneOf: [
+        { properties: { origin: { type: 'string' } } },
+        {
+          properties: {
+            origins: {
+              type: 'array',
+              items: { type: 'string' },
+            },
+          },
+        },
+      ],
```

Comment:
As I test locally, in this form schema will in most cases fail, we've used `oneOf`, and no property listed is required (so in most cases both variants match, and that's invalid by schema).

I guess it's not exposed by tests, only because there are no tests that involve this event written with `runServerless`

Normally we configure such things, by defining involved properties normally in `properties` section , and use `oneOf` to configure two different `required` collections